desktop_server/src/com/mygdx/game/server/network/gameinstance/services/MoveEntityService.java

```diff
@@ -20,9 +22,30 @@ public void moveEntity(int entityId, int x, int y, GameRoom room) {
     var world = room.getGameInstance().getWorld();
     var coordinatesMapper = world.getMapper(Coordinates.class);
     var destination = coordinatesMapper.get(entityId);
-    destination.setCoordinates(x, y);
-    log.info("Send position component");
-    setDirty(entityId, Coordinates.class, world);
-    world.process();
+
+    var distance = DistanceUtil.distance(destination, new Coordinates(x, y));
+
+    var moveRangeMapper = world.getMapper(MoveRange.class);
+    var range = moveRangeMapper.get(entityId);
+    var currentRange = range.getCurrentRange();
+    log.info("Server - movement service");
+    log.info("Current range: " + currentRange + "\nMovement distance: " + distance);
+
+    if (currentRange < distance) {
+      log.warning("Position out of move range!!!");
+    }
+    else{
+      destination.setCoordinates(x, y);
+      var updatedRange = currentRange - distance;
+      range.setCurrentRange(updatedRange);
+
+      log.info("Send position component");
+      log.info("Unit has " + updatedRange + " move points left");
+      setDirty(entityId, Coordinates.class, world);
+//      TODO: Coś takiego powinienem zrobić? Czy o co chodzi z tym DirtyComponent???
```

Comment:
To też jest coś podobnego w desktop_serwer/AttackentityService w #128 może pomoże

Reply:
tak, o to chodzi z tym dirtycomponent